File: UnitTests/Samples/LateBreaking/ListTimeZoneAndConvert/VB/TimeZoneConverterVB/TZConv.vb
Imports System.Threading

'---------------------------------------------------------------------
'  This file is part of the Microsoft .NET Framework SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------
''' <summary>
''' simple form to show the usage of the TimeZone and the TimeZoneCollection classes 
''' </summary>
''' <remarks></remarks>
Public Class TZConv

    Private Sub TZConv_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        Dim tzs As New TimeZoneList
        Dim i As Integer

        For i = 0 To tzs.Count - 1
            Me.CmbDestTZ.Items.Add(tzs.Item(i))

            Debug.Print(tzs.Item(i).DisplayName)

        Next
        Me.TxtSourceDate.Text = Now.ToString(Thread.CurrentThread.CurrentCulture)
    End Sub

    Private Sub BtnExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnExit.Click
        Me.Close()
    End Sub

    Private Sub BtnConvert_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnConvert.Click

        ' check if a target TimeZone has been selected
        If IsNothing(Me.CmbDestTZ.SelectedItem) Then
            MessageBox.Show("Please select a target timezone", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button1, MessageBoxOptions.DefaultDesktopOnly, False)
        Else
            Call DoConvert()
        End If

    End Sub

    ''' <summary>
    ''' Executes date conversion based on selected Source and Destination TimeZones
    ''' </summary>
    ''' <remarks></remarks>
    Private Sub DoConvert()

        Dim d, UTCd As DateTime
        Try
            d = DateTime.Parse(TxtSourceDate.Text, Thread.CurrentThread.CurrentCulture)
        Catch ex As FormatException
            MessageBox.Show(ex.Message + " - Please enter a valid date", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button1, MessageBoxOptions.DefaultDesktopOnly, False)
            Exit Sub
        End Try

        ' get the UTC date/time equivalent of the Source Date and show it
        UTCd = d.ToUniversalTime()
        Me.lblUTCDate.Text = UTCd.ToString(Thread.CurrentThread.CurrentCulture)

        ' get the Destination Bias (number of minutes from UTC)
        Dim DestBias As Integer
        DestBias = CType(Me.CmbDestTZ.SelectedItem, TimeZone).Bias

        ' Add difference between timezones to specified date
        Me.LblDestDate.Text = UTCd.AddMinutes(DestBias).ToString(Thread.CurrentThread.CurrentCulture)
    End Sub

End Class

File: UnitTests/Samples/LateBreaking/ListTimeZoneAndConvert/CS/TimeZoneConvertCS/TZConv.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.Threading;

namespace Microsoft.Samples.TimeZoneConvertCS
{
	partial class TZConv : Form
	{
		public TZConv()
		{
			InitializeComponent();
		}

		private void btnExit_Click(object sender, EventArgs e)
		{
			this.Close();
		}

		private void TZConv_Load(object sender, EventArgs e)
		{
			
			TimeZoneList tzs = new TimeZoneList();

			int i;

			for (i=1; i < tzs.Count; i++)
            {
				this.cmbDestTZ.Items.Add(tzs[i]);
			}
            this.txtSourceDate.Text = DateTime.Now.ToString(Thread.CurrentThread.CurrentCulture);
		}

		private void btnConvert_Click(object sender, EventArgs e)
		{
            // Check if the target TimeZone was selected
            if (this.cmbDestTZ.SelectedItem == null)
                MessageBox.Show("Please select a target timezone", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button1, MessageBoxOptions.DefaultDesktopOnly, false);
            else 
			    DoConvert();
		}

		private void DoConvert()
		{
			DateTime d, UTCd;
            try 
            {
                d = DateTime.Parse(txtSourceDate.Text, Thread.CurrentThread.CurrentCulture);
            }
			catch (FormatException Ex) 
            {
                MessageBox.Show(Ex.Message + " - Please enter a valid date", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button1, MessageBoxOptions.DefaultDesktopOnly, false);
			    return;
            }

            // get the UTC date/time equivalent to the SourceDate 
            UTCd = d.ToUniversalTime();
            this.lblUTCDate.Text = UTCd.ToString(Thread.CurrentThread.CurrentCulture);

			// get the Destination Bias (number of minutes from UTC)
			int DestBias;
			DestBias = ((TimeZone)this.cmbDestTZ.SelectedItem).Bias;

			// Add the Destination Bias to the UTC date/time
            this.lblDestDate.Text = UTCd.AddMinutes(DestBias).ToString(Thread.CurrentThread.CurrentCulture);
        }
	}
}